server/src/main/java/org/opensearch/extensions/ExtensionActionListenerOnFailureRequest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * CLusterService Request for Action Listener onFailure
+ *
+ * @opensearch.internal
+ */
+public class ExtensionActionListenerOnFailureRequest extends TransportRequest {
+    private static final Logger logger = LogManager.getLogger(ExtensionRequest.class);
+    private String failureException;
```

Comment:
Nit: an Exception is not a String. Can we name this what it's expected to be, like `failureExceptionMessage` if the expected input is from `getMessage()`, or `failureExceptionClassName`/`failureExceptionType` if it's just the type?  

Whichever string is expected should be explained in the constructor's javadocs right below this.

Oh, wait, there are no javadocs. :-)